src/gui/mainwindow.cpp

```diff
@@ -126,6 +117,7 @@ MainWindow::MainWindow(QWidget *parent)
 #if defined(Q_OS_WIN) || defined(Q_OS_MAC)
     , m_wasUpdateCheckEnabled(false)
 #endif
+    , m_searchEngine(new SearchEngine)
```

Comment:
This frustrates me too. But I think that mentioned earlier intent ti get notification handlers out of the manager classes will solve this problem.